### ISSUE TITLE
getting java.lang.NullPointerException on versions >5.17.17

### ISSUE THREAD

**Issue by mjok** [OpenHFT/Chronicle-Core#94]
as in my case Chronicle queue loaded by `javaagent` by Bootstrap Class Loader  Jvm.class.getClassLoader() return null causing NPE.
This is related to #86.

```
Exception in thread "main" java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:386)
        at sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:401)
Caused by: java.lang.ExceptionInInitializerError
        at net.openhft.chronicle.core.cleaner.CleanerServiceLocator.isAllowedInThisMajorVersion(CleanerServiceLocator.java:54)
        at net.openhft.chronicle.core.cleaner.CleanerServiceLocator.cleanerService(CleanerServiceLocator.java:28)
        at net.openhft.chronicle.bytes.NativeBytesStore.<clinit>(NativeBytesStore.java:44)
        at net.openhft.chronicle.bytes.BytesStore.from(BytesStore.java:52)
        at net.openhft.chronicle.wire.WireType.<clinit>(WireType.java:328)
        at net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder.binary(SingleChronicleQueueBuilder.java:214)
        at net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder.binary(SingleChronicleQueueBuilder.java:210)
        at net.openhft.chronicle.queue.ChronicleQueue.singleBuilder(ChronicleQueue.java:102)
        at com.jkoolcloud.remora.core.output.ChronicleOutput.init(ChronicleOutput.java:80)
        at com.jkoolcloud.remora.core.output.OutputManager.install(OutputManager.java:67)
        at com.jkoolcloud.remora.core.output.OutputManager.<init>(OutputManager.java:44)
        at com.jkoolcloud.remora.core.output.OutputManager.<clinit>(OutputManager.java:33)
        at com.jkoolcloud.remora.Remora.premain(Remora.java:82)
        ... 6 more
Caused by: java.lang.NullPointerException
        at net.openhft.chronicle.core.Jvm.loadSystemProperties(Jvm.java:137)
        at net.openhft.chronicle.core.Jvm.<clinit>(Jvm.java:132)
        ... 19 more
```

**Comment by RobAustin** [comment]
what version of java are you using

**Comment by mjok** [comment]
Oracle 1.8

**Comment by RobAustin** [comment]
This is the line of code that produces the NPE 

```
            InputStream is0 = Jvm.class.getClassLoader().getResourceAsStream(name);
```

so either

```
Jvm.class.getClassLoader() == null
```

or perhaps 

```
name == null
```

could you run it in a debugger and let me know what was null

Thanks

**Comment by RobAustin** [comment]
I've just re-read your error report and see that you said "Jvm.class.getClassLoader() is null " are you sure that it is this that is null ?

**Comment by mjok** [comment]
I have checked it on debugger, it's null.

**Comment by RobAustin** [comment]
@mjok are you able to make a change to this code 

```
net.openhft.chronicle.core.Jvm.loadSystemProperties(Jvm.java:137)
```

to handle the case where getClassLoader() returns null, due to it being the bootstrap classloader.

if you could make the change and create a pull request we can merge in the fix to this issue. Otherwise, we will get around to looking at it sometime later, however this is likely to take a while as it is not a high priority issue. Also note:  there may be other places in the code which assume that getClassLoader() is not returning null which may also require fixing which will also have to be addressed.

**Comment by mjok** [comment]
I'll do that. Thanks

**Comment by hft-team-city** [comment]
Released in [Chronicle-Core-2.19.0](https://github.com/OpenHFT/Chronicle-Core/releases/chronicle-core-2.19.0), [BOM-2.19.4](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.19.4)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Core-2.20.102](https://github.com/OpenHFT/Chronicle-Core/releases/chronicle-core-2.20.102), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
